lib/user-interface/react-app/src/components/chatbot/multi-chat-input-panel.tsx

```diff
@@ -0,0 +1,261 @@
+import {
+  Button,
+  Select,
+  SpaceBetween,
+  StatusIndicator,
+  SelectProps,
+  Container,
+  Spinner,
+  Icon,
+  Toggle,
+} from "@cloudscape-design/components";
+import { useContext, useEffect, useState } from "react";
+import { ReadyState } from "react-use-websocket";
+import SpeechRecognition, {
+  useSpeechRecognition,
+} from "react-speech-recognition";
+
+import { AppContext } from "../../common/app-context";
+import TextareaAutosize from "react-textarea-autosize";
+import { OptionsHelper } from "../../common/helpers/options-helper";
+import { useNavigate } from "react-router-dom";
+import { StorageHelper } from "../../common/helpers/storage-helper";
+import styles from "../../styles/chat.module.scss";
+import { WorkspaceItem } from "../../common/types";
+import { ChatInputState } from "./types";
+
+export interface MultiChatInputPanelProps {
+  running: boolean;
+  readyState: ReadyState;
+  onSendMessage: (msg: string) => void;
+  enabled: boolean;
+  workspaces: WorkspaceItem[];
+  showMetadata?: boolean;
+  onChange: (value: boolean) => void;
+}
+
+export abstract class ChatScrollState {
+  static userHasScrolled = false;
+  static skipNextScrollEvent = false;
+  static skipNextHistoryUpdate = false;
+}
+
+const workspaceDefaultOptions: SelectProps.Option[] = [
+  {
+    label: "No workspace (RAG data source)",
+    value: "",
+    iconName: "close",
+  },
+  {
+    label: "Create new workspace",
+    value: "__create__",
+    iconName: "add-plus",
+  },
+];
+
+export default function MultiChatInputPanel(props: MultiChatInputPanelProps) {
+  const appContext = useContext(AppContext);
+  const navigate = useNavigate();
+  const { transcript, listening, browserSupportsSpeechRecognition } =
+    useSpeechRecognition();
+  const [state, setState] = useState<ChatInputState>({
+    value: "",
+    selectedModel: null,
+    selectedWorkspace: workspaceDefaultOptions[0],
+    modelsStatus: "loading",
+    workspacesStatus: "loading",
+  });
+
+  useEffect(() => {
+    if (transcript) {
+      setState((state) => ({ ...state, value: transcript }));
+    }
+  }, [transcript]);
+
+  useEffect(() => {
+    if (!appContext) return;
+  }, [appContext, state.modelsStatus]);
+
+  useEffect(() => {
+    const onWindowScroll = () => {
+      if (ChatScrollState.skipNextScrollEvent) {
+        ChatScrollState.skipNextScrollEvent = false;
+        return;
+      }
+
+      const isScrollToTheEnd =
+        Math.abs(
+          window.innerHeight +
+            window.scrollY -
+            document.documentElement.scrollHeight
+        ) <= 10;
+
+      if (!isScrollToTheEnd) {
+        ChatScrollState.userHasScrolled = true;
+      } else {
+        ChatScrollState.userHasScrolled = false;
+      }
+    };
+
+    window.addEventListener("scroll", onWindowScroll);
+
+    return () => {
+      window.removeEventListener("scroll", onWindowScroll);
+    };
+  }, []);
+
+  //   useLayoutEffect(() => {
```

Comment:
ditto

Reply:
I remove this since the UI behaves correctly an you never get history passed as prop to this panel